adapters/httpapi_compact.c

```diff
@@ -235,7 +235,7 @@ HTTP_HANDLE HTTPAPI_CreateConnection(const char* hostName)
             tlsio_config.port = 443;
             tlsio_config.underlying_io_interface = NULL;
             tlsio_config.underlying_io_parameters = NULL;
-
+            tlsio_config.protocol="http";
```

Comment:
Hey @maallahatem , thank you very much for the PR. 
I have a few points about the code change though.
1. The TLS layer should not be tied to any specific protocol, so the best way would be create a flag in the tlsio_config structure like "invoke_on_send_complete_callback_for_fragments" (true/false);
2. I am concerned about the behavior of the upper layer if it now starts firing multiple `on_send_complete` for a single send (single from an upper layer perspective) - I will need to test to make sure this is not a breaking change for the SDK;
3. I need to double check if something need to be done on the other tlsio_* adapters for this new flag, in case they also need to consume it or document that they ignore it completely. 

For now if you could address item 1 above, I'll work on items 2 and 3 within the next week. 

Thanks!

Reply:
Thanks @maallahatem , I see you made the changes. We will verify them next.